File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import java.util.List;
import java.util.Map;

import com.pinyougou.pojo.TbBrand;

import entity.PageResult;

/**
 * 品牌接口
 * 
 * @author ybn
 * @param <TbBrand>
 *
 */
public interface BrandService {

	public List<TbBrand> findAll();

	/**
	 * 品牌分页
	 * 
	 * @param pageNum
	 *            当前页码
	 * @param pageSize
	 *            每页记录数
	 * @return
	 */
	public PageResult findPage(int pageNum, int pageSize);

	/**
	 * 添加
	 * 
	 * @param tbBrand
	 */
	public void add(TbBrand Brand);

	/**
	 * 根据id查找单独一个
	 */

	public TbBrand findOne(long id);

	/**
	 * 修改
	 */
	public void update(TbBrand brand);

	/**
	 * 删除
	 */
	public void delete(long[] ids);

	/**
	 * 品牌分页
	 * 
	 * @param pageNum
	 *            当前页码
	 * @param pageSize
	 *            每页记录数
	 * @return
	 */
	public PageResult findPage(TbBrand brand, int pageNum, int pageSize);

	public List<Map> selectOptionList();
}

File: pinyougou-manager-web/src/main/webapp/js/service/brandService.js
app.service("brandService", function ($http) {
    //查找所有
    this.findAll = function () {
        return $http.get('../brand/findAll.do');
    };
    //分页
    this.findPage = function (page,rows) {
        return $http.get('../brand/findPage.do?page=' + page + '&size=' + size);
    };
    //查找一个
    this.findOne = function (id) {
        return $http.get('../brand/findOne.do?id=' + id);
    };
    //添加
    this.add = function (entity) {
        return $http.post('../brand/add.do', entity)
    }
    //修改
    this.update = function (entity) {
        return $http.post('../brand/update.do', entity)
    }
    //删除
    this.dele = function (ids) {
        return $http.get('../brand/delete.do?ids=' + ids);
    }
    //搜索
    this.search = function (page, size, searchEntity) {
        return $http.post('../brand/search.do?page=' + page + '&size=' + size, searchEntity);
    }
    //下拉列表数据
    this.selectOptionList=function(){
        return $http.get('../brand/selectOptionList.do');
    }
});